### PR TITLE
Issue 6081: Added JNI thread attach/detach struct to dmSDK

### LINKED ISSUE #6081: Add JNI thread attacher to Defold SDK
**Is your feature request related to a problem? Please describe (REQUIRED):**
It is important to detach a thread from the JNI to not leak any resources or lock up the system. Using a ThreadAttacher struct with a constructor to attach a thread to the JNI and a destructor to detach the thread would guarantee that the thread is detached.

**Describe the solution you'd like (REQUIRED):**
The Defold SDK should provide a thread attach/detach mechanism to help extension creators detach from the JNI properly

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/dlib/src/dlib/android.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            ok = false;
+        }
+        m_Env->ExceptionClear();
+        m_Activity->vm->DetachCurrentThread();
+        m_IsAttached = false;
+    }
+    return ok;
+}
+
+jclass LoadClass(JNIEnv* env, const char* class_name)
```
Comment (JCash) [inline engine/dlib/src/dlib/android.cpp:53]: Regarding our previous discussion about what other utility functions we should add.
This was a use case that felt like a good first step to centralize.

File: engine/dlib/src/dlib/android.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            m_Env->ExceptionDescribe();
+            ok = false;
+        }
+        m_Env->ExceptionClear();
+        m_Activity->vm->DetachCurrentThread();
+        m_IsAttached = false;
+    }
+    return ok;
+}
```
Comment (JCash) [inline engine/dlib/src/dlib/android.cpp:51]: Although the `Detach()`is called automatically, there are also cases when you wish to detach and check there were any errors.

File: engine/dlib/src/dmsdk/dlib/android.h [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+/*# SDK Android API documentation
+ * Android utility functions
+ *
+ * @document
+ * @name Android
+ * @namespace dmAndroid
+ * @path engine/dlib/src/dmsdk/dlib/android.h
+ */
+
+namespace dmAndroid {
```
Comment (JCash) [inline engine/dlib/src/dmsdk/dlib/android.h:30]: Added a new dmsdk header, and a new namespace.

Comment (AGulev) [reply engine/dlib/src/dmsdk/dlib/android.h:30]: I love it!!! 🥰

**Review by AGulev — APPROVED** [review]
I love it!!! 🥰

File: engine/dlib/src/dlib/android.cpp (hunk 1 above)
Comment (britzl) [reply engine/dlib/src/dlib/android.cpp:53]: Yes, this is excellent. It will remove a bunch of code in several extensions!

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**